docs/cookbook/getFirstTuesdayOfMonth.mjs

```diff
@@ -0,0 +1,30 @@
+import assert from "assert";
+
+/**
+ * Gets the first tuesday of the month and returns its date
+ *
+ * @param {Temporal.YearMonth} queriedMonth YearMonth instance to query
+ * @returns {Temporal.DateTime} DateTime Instance which gives first tuesday
+ */
+function getFirstTuesday(queriedDate) {
+    // We first need to convert to a date
+    let date = Temporal.DateTime.from({
+        year: queriedDate.year,
+        month: queriedDate.month,
+        day: 1
+    });
+
+    // Start at the beginning of the month then stop at the first tuesday
+    while (date.dayOfWeek !== 2) {
+        date = date.plus({ days: 1 });
+    }
```

Comment:
I think this can just be `date = date.plus({ days: 7 - date.dayOfWeek + 2 })`? Or something similar?

Reply:
could you enlighten me here, i don't see how this would work if say the first day of the month Monday was the 1st. So then we're looking for the 2nd (Tuesday) but would end up skipping over it.
1st + (7 - 1 + 2)

Reply:
I had something like that for `Date`; I can't remember now. It might have worked because of the numbering `Date` uses for the result of `getDay()` though, so it might not work for `Temporal`... Maybe `Math.abs(7 - date.dayOfWeek) + 3`? You know what? What you have works, pay no attention to me :).

Reply:
I think `date.plus({ days: (9 - date.dayOfWeek) % 7 })` ought to work. We have Monday = 1, Sunday = 7, and we want to add a positive number smaller than 7 to get to the first Tuesday. If we're already on a Tuesday (2) then we want to add 0. So for the first of the month being a Monday through Sunday the additions are: 1, 0, 6, 5, 4, 3, 2 which is given by that formula.

Alternatively making it more explicit via a lookup table might look less mystical to the reader: `[1, 0, 6, 5, 4, 3, 2][date.dayOfWeek - 1]`